lib/run-status.js

```diff
@@ -123,6 +125,10 @@ RunStatus.prototype.handleTest = function (test) {
 		this.errors.push(test);
 	}
 
+	if (test.failing && !test.error) {
+		this.knownFailures.push(test);
```

Comment:
@cgcgbcbc, @nfcampos is running into that as well: https://github.com/avajs/ava/pull/844#issuecomment-220554007.

Maybe leave that for now, focus on the reporter tests (which use stubbed data anyway).